x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.ts

```diff
@@ -59,10 +59,13 @@ export const fetchIndices = async (
   client: IScopedClusterClient,
   indexPattern: string,
   returnHiddenIndices: boolean,
-  includeAliases: boolean
+  includeAliases: boolean,
+  alwaysShowSearchPattern?: 'search-'
```

Comment:
I'm not going to hold up this PR over this, but given the fourth parameter and the conditional complexity in this function, I think it's very clear that this function is trying to do too many things at the same time and we should split this up into to different functions, one for each endpoint.

Reply:
100% agreed. I have created a ticket for that already. We should start working on this ASAP. I am willing to